src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -41,51 +41,107 @@ const EditBootableVolumesModal: FC<EditBootableVolumesModalProps> = ({
     () => Object.keys(convertResourceArrayToMap(preferences)).sort((a, b) => a.localeCompare(b)),
     [preferences],
   );
-  const initialMetadata: BootableVolumeMetadata = useMemo(() => {
+
+  const instanceTypesToSizesMap = useMemo(
+    () => getInstanceTypesToSizesMap(instanceTypesNames),
+    [instanceTypesNames],
+  );
+
+  const initialParams = useMemo(() => {
+    const instanceTypeLabel =
+      dataSource?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]?.split('.');
+
     return {
-      labels: dataSource?.metadata?.labels,
-      annotations: dataSource?.metadata?.annotations,
+      preference: dataSource?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
+      instanceType: instanceTypeLabel?.[0],
+      size: instanceTypeLabel?.[1],
+      description: dataSource?.metadata?.annotations?.[ANNOTATIONS.description],
     };
   }, [dataSource]);
-  const [metadata, setMetadata] = useState<BootableVolumeMetadata>(initialMetadata);
 
-  const setBootableVolumeMetadata = (key: string, fieldKey: string) => (value: string) =>
-    setMetadata((prevState) => ({
-      ...prevState,
-      [key]: { ...prevState[key], [fieldKey]: value },
-    }));
+  const [preference, setPreference] = useState<string>(initialParams.preference);
+  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType);
+  const [size, setSize] = useState<string>(initialParams.size);
+  const [description, setDescription] = useState<string>(initialParams.description);
+
+  const sizeOptions = useMemo(
+    () => instanceTypesToSizesMap[instanceType] || [],
+    [instanceType, instanceTypesToSizesMap],
+  );
+
+  const onInstanceTypeChange = (instanceTypeName: string) => {
+    setInstanceType(instanceTypeName);
+    setSize(instanceTypesToSizesMap[instanceTypeName][0]);
+  };
+
+  const onChangeVolumeParams = () => {
```

Comment:
Just a note, any functions passed to React components as props (e.g. callbacks) should be referentially stable across re-renders. Otherwise, you're creating a new function instance on every render here, passing it to other React components and causing re-render of these components.

This can be done via `React.useCallback` hook https://reactjs.org/docs/hooks-reference.html#usecallback

On the other hand, the extra re-renders aren't normally a big issue, so developers may opt into not doing this.

Reply:
`React.useCallback` was added in some of the followup PRs.